core/arch/aarchxx/mangle.c

```diff
@@ -3391,6 +3427,10 @@ mangle_exclusive_load(dcontext_t *dcontext, instrlist_t *ilist, instr_t *instr,
                 instr_create_restore_from_tls(dcontext, swap_reg, swap_slot));
         }
     }
+    if (xzr_restore)
```

Comment:
Not at L3325.  Possibly at 3370, but not for the A64 base==sp case: so it gets complicated.  I don't think it saves anything to do it early: the slot count is fixed (these are DR's core scratch slots) and we are not out of them here (we do run out for a case in #5247).  So it would add complexity for no gain: so I would prefer to leave it.